File: rotate.java
public class Solution {
    public void rotate(int[][] matrix) {
        int size = matrix.length;
        if (size < 1)
           return;
        int temp = 0;
        for (int i = 0; i < size/2; i++){
            for (int j = 0; j < size; j++){
                temp = matrix[i][j];
                matrix[i][j] = matrix[size-1-i][j];
                matrix[size-1-i][j] = temp;
            }
        }
        
        for (int i =0; i <size-1; i++){
            for (int j = i+1; j< size; j++){
                temp = matrix[j][i];
                matrix[j][i] = matrix[i][j];
                matrix[i][j] = temp;
            }
        
        }
        
    }
}

File: rotate.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        if len(nums) < 2 or k%len(nums) == 0:
            return 
        
        size = len(nums)
        k = k%size
        cur = start = size-1
        temp = nums[start]
        for i in range(size):
            if cur - k >= 0:
                nums[cur] = nums[cur-k]
                cur -= k
            else:
                if size+cur-k == start:
                    nums[cur] = temp
                    cur = start = start -1
                    temp = nums[start]
                else:
                    nums[cur] = nums[cur-k]
                    cur = size+cur-k

        nums[cur] = temp
        
        
        
        
